filebeat/input/kafka/config.go

```diff
@@ -0,0 +1,220 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kafka
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/Shopify/sarama"
+
+	"github.com/elastic/beats/libbeat/common/kafka"
+	"github.com/elastic/beats/libbeat/common/transport/tlscommon"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/monitoring"
+	"github.com/elastic/beats/libbeat/monitoring/adapter"
+	"github.com/elastic/beats/libbeat/outputs"
+)
+
+type kafkaInputConfig struct {
+	// Kafka hosts with port, e.g. "localhost:9092"
+	Hosts         []string          `config:"hosts" validate:"required"`
+	Topics        []string          `config:"topics" validate:"required"`
+	GroupID       string            `config:"group_id" validate:"required"`
+	ClientID      string            `config:"client_id"`
+	Version       kafka.Version     `config:"version"`
+	InitialOffset initialOffset     `config:"initial_offset"`
+	RetryBackoff  time.Duration     `config:"retry_backoff" validate:"min=0"`
+	MaxWaitTime   time.Duration     `config:"max_wait_time"`
+	Fetch         *kafkaFetch       `config:"fetch"`
+	Rebalance     *kafkaRebalance   `config:"rebalance"`
+	TLS           *tlscommon.Config `config:"ssl"`
+	Username      string            `config:"username"`
+	Password      string            `config:"password"`
+}
```

Comment:
nit: I noticed `Fetch` and `Rebalance` are already set in the default config. In this case there is no reason to use pointers. Code normally uses pointer to check if a config (section) is fully optional, and we don't even care about defaults if it not given at all.